source/MaterialXCore/Element.cpp

```diff
@@ -482,6 +545,107 @@ TypeDefPtr TypedElement::getTypeDef() const
 // ValueElement methods
 //
 
+bool ValueElement::isEquivalent(ConstElementPtr rhs, ElementEquivalenceOptions& options) const
+{
+    if (getCategory() != rhs->getCategory() ||
```

Comment:
Would be nice if we could find the right way to have less code duplication between these two functions.

Reply:
I think a virtual `isAttributeEquivalent(elementPtr, attrbuteString)` can work here so there is one `Element::isEquivalent()`. The no-op child element check is a small overhead that is part of the ValueElement compare but it's not more than what already occurs with `operator==` and will keep it generic in case ValueElements ever have children.